src/chain_sync/chain_muxer.rs

```diff
@@ -1002,3 +1012,16 @@ where
         }
     }
 }
+
+// The formula matches lotus
+// ```go
+// sinceGenesis := build.Clock.Now().Sub(genesisTime)
+// expectedHeight := int64(sinceGenesis.Seconds()) / int64(build.BlockDelaySecs)
+// ```
+// See <https://github.com/filecoin-project/lotus/blob/b27c861485695d3f5bb92bcb281abc95f4d90fb6/chain/sync.go#L180>
+fn get_now_epoch(genesis_timestamp: i64, block_delay: i64) -> i64 {
```

Comment:
nit: If we pass in the timestamp, this function would be pure. Pure functions are nice.

Reply:
Fixed.